schemas/xml/README.md

```diff
@@ -1,57 +1,449 @@
+There is thus no need to include the discriminator in the serialization, and a redundant discriminator would only clutter the XML document.
+
+We therefore distinguish two serializations of instances: one with discriminator, and one without, respectively.
+
+#### Instances Serialized with Discriminator
+
+We use the XML element named according to the concrete class at the serialization as the discriminator.
+The attributes of the instance are nested below this discriminator XML element.
+
+Let us make an example.
+The example will be agnostic of the particular meta-model version, so that it can age well across different versions.
+We fantasize a meta-model class `SomeAbstractClass` and assume it has two descendant classes, `SomeConcreteClass` and `AnotherConcreteClass`.
+Let us assume yet another class, `YetAnotherClass`, entails the attribute `someAttribute` of type `SomeAbstractClass`.
+
+Here is how the XML structure corresponding to `YetAnotherClass/someAttribute` would look like in this fantasized case, where the value is an instance of `SomeConcreteClass`:
+
+```xml
+
+<someAttribute>
+    <SomeConcreteClass>
+        ...
+        serialized
+        attributes
+        of
+        SomeConcreteClass
+        ...
+    </SomeConcreteClass>
+</someAttribute>
+```
+
+If the value is an instance of `AnotherConcreteClass`, the serialization becomes:
+
+```xml
+
+<someAttribute>
+    <AnotherConcreteClass>
+        <!-- ... serialized attributes of AnotherConcreteClass ... -->
+    </AnotherConcreteClass>
+</someAttribute>
+```
+
+The abstract class, `SomeAbstractClass`, does not show up in the serialization at all, as it is redundant information.
+
+#### Instances Serialized without Discriminator
+
+If the concrete type of the attribute at deserialization is unambiguous by the meta-model, we omit the discriminator to reduce the clutter.
+The instance is simply serialized as a sequence of XML elements corresponding to its attributes.
+
+Let us fantasize yet another example, similar to the one in Section [Instances Serialized with Discriminator].
+We will again draw an example such that it is agnostic of meta-model version for better evolution of this Readme.
+Assume a class `SomeClass` with an attribute `SomeClass/someAttribute`.
+Now imagine the type of `someAttribute` to be the class `AnotherClass`.
+The class `AnotherClass` provides attributes `AnotherClass/anotherAttribute` and `AnotherClass/yetAnotherAttribute`.
+The class `AnotherClass` has no descendants, so the concrete type of `SomeClass/someAttribute` is unambiguous.
+
+[Instances Serialized with Discriminator]: #Instances-Serialized-with-Discriminator
+
+Here is how the XML structure would look like:
+
+```xml
+
+<someAttribute>
+    <anotherAttribute>
+        <!-- ... -->
+    </anotherAttribute>
+    <yetAnotherAttribute>
+        <!-- ... -->
+    </yetAnotherAttribute>
+</someAttribute>
+```
+
```

Comment:
This means we are not backward compatible if we would introduce inheritance in a later version? Introducing inheritance on UML level would be backward compatible.